libp2p-swarm/src/swarm.rs

```diff
@@ -233,7 +235,7 @@ impl<T, C, H, If, F> Future for SwarmFuture<T, C, H, F>
                     self.listeners.push(listener);
                 },
                 Ok(Async::Ready(None)) => {},
-                Err(err) => return Err(err),
+                Err(_err) => {},        // Ignoring errors
```

Comment:
why are errors being ignored here?

Reply:
An error happens if the remote sends us bad data, so we just drop the connection. This probably needs a comment and a log entry.